src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -287,6 +287,8 @@ export class SearchSource {
     // This still uses bfetch for batching.
     if (!options?.strategy && syncSearchByDefault) {
       options.strategy = ES_SEARCH_STRATEGY;
+      // `ES_SEARCH_STRATEGY` doesn't support search sessions, hence remove sessionId
+      options.sessionId = undefined;
```

Comment:
There are a lot of ways how to handle this, but since this fix is focused on `courier:batchSearches`, I suggest we contain the fix together with the option usage.

Long term, I think we find a how-to avoid search session overhead (both on UI and server) when running a search strategy that doesn't support search sessions. (see pr description for more details)

Reply:
Won't this disable client side caching of responses as well 🤔

Reply:
Yes @lizozom! 

But otherwise not sure how to solve it also keeping the fix surface as small as possible: 

We want cache to still work, so we need to pass `sessionId`, 
but then we also don't want to `trackSearch` in client-side session service and ideally don't want to pass sessionId to the server. 

I don't think we should put such conditional logic that depends on a specific strategy into client-side search interceptor, hence I suggest going with removing sessionId in searchSource level